pkg/apis/core/v1alpha1/bindingphase.go

```diff
@@ -18,12 +18,14 @@ limitations under the License.
 
 package v1alpha1
 
+import "encoding/json"
+
 // BindingState is to identify the current binding status of given resources
 type BindingState int
 
 // MarshalJSON returns a JSON representation of a BindingState.
 func (s BindingState) MarshalJSON() ([]byte, error) {
-	return []byte(s.String()), nil
+	return json.Marshal(s.String())
```

Comment:
@negz were you able to walk through this change in a full/real environment?  did the logs show any other errors like the following ones I encountered the other day?
```
E0312 09:56:37.632389       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.CloudsqlInstance: v1alpha1.CloudsqlInstanceList.Items: []v1alpha1.CloudsqlInstance: v1alpha1.CloudsqlInstance.Status: v1alpha1.CloudsqlInstanceStatus.BindingStatusPhase: Phase: readUint64: unexpected character: �, error found in #10 byte of ...|ngPhase":"Bound","en|..., bigger context ...|,"Status":"True","Type":"Ready"}],"bindingPhase":"Bound","endpoint":"35.236.115.81","instanceName":"|...
```

```
2019/03/12 09:56:37 failed to update status of instance postgresql-e7304048-44a4-11e9-b6e3-080027d9d28d: failed to update status of CRD instance postgresql-e7304048-44a4-11e9-b6e3-080027d9d28d: Operation cannot be fulfilled on cloudsqlinstances.database.gcp.crossplane.io "postgresql-e7304048-44a4-11e9-b6e3-080027d9d28d": the object has been modified; please apply your changes to the latest version and try again
```

do we need to implement an `UnmarshalJSON` function also?

Reply:
Good point - we do. I've implemented that and will test it now.